### DISCUSSION TITLE
ClickHouse Support

### DISCUSSION THREAD

**Discussion by Slach** [RamiAwar/dataline#316] [Ideas]
Thanks for your great tool, it help me a lot when explore data in MySQL

Could you add support for ClickHouse?

ClickHouse is blazing fast OLAP database which require opensource tool for data exploration,

Look https://clickhouse.com/docs/en/integrations/language-clients/javascript for details about TS integratoins

**Comment by RamiAwar** [comment]
Hey @Slach! I see this isn't trivial (not yet supported in the Langchain SQL Database class that we use).

It might be easy, but will need someone to come and try it. There exists an SQLAlchemy connector already for clickhouse (you have to integrate it in our backend first (Python)).

Feel free to give it a shot, it should be easy to see if it works or not! All we need is the schema + ability to execute SQL.

If you wanna wait for us to implement it, I think it's one of the much lower priority items so might take a while.. 
No one else asked for this yet but I'd love to support it and help you do this!
